docs/guides/data/mounting_data.md

```diff
@@ -1,105 +1,192 @@
+underlying data.
+
+### Mounting a Dataset
+
+#### What does it mean to "mount" a dataset to a job?
+In the world of file systems, the term "mount" means to attach one file system
+or folder to another file system. For example, when you insert a flash drive
+(which is a mini file system) into your computer, its file system gets mounted
+to your computer's main file system so that you can retrieve, remove, or save
+data to the flash drive. Once the flash drive is mounted to your computer's
+file system, other programs can access it as if it were a native part of your
+computer's file system. This same mounting pattern is how datasets are handled
+on FloydHub.
+
+When you use the `floyd run` command to run a job, your code will be sent up to
+FloydHub and run on a powerful deep-learning server that can run your job. If
+you want your code to be able to access a dataset during the job, you need to
+mount the dataset to the server where the job is running (just like you need to
+plug a flash drive into your computer in order to access its files). Mounting
+datasets to a job is easy: just pass the `--data` flag to the `floyd run` cli
+command as detailed below.
+
+### The `--data` flag
+
+To properly use the `--data` flag with `floyd run`, you need to specify two
+things:
+
+  1. The name of the dataset you want to mount.
+  2. A name for the folder where the data will be
+  accessible to your code during the job, we call this the "mount point". You
+  can give this folder (mount point) any name you want. Don't precede this name
+  with a forward slash `/`.
+
+These two things are separated by a `:` with no spaces. Here's the syntax:
 
-## Datasets
-
-Floyd datasets are directories of data files that can be used in a project run. To create a new dataset, please follow 
-[this guide](../basics/create_new/#create-a-new-dataset). You can view the datasets you 
-have created in the [datasets page](https://www.floydhub.com/datasets) in the dashboard. You can also view 
-public datasets by searching for it on FloydHub.
-
-## Mounting datasets
+```
+floyd run --data <data_set_name>:<mount_point> …(rest of run command)
+```
 
-### Overview
+Let's go through a couple of examples to show how to mount one or more datasets to your job.
 
-You can mount one or more datasets when you run a job using the [floyd run](../../commands/run) command. 
-This automatically makes the mounted datasets available for you to use in your job, without having to download them each time.
+#### Example 1
 
-To mount a specific version of a dataset, you need to specify its [full name](#datasource-name) and the [mount point](#mount-point). 
-The syntax is `--data <data_name>:<mount_point>`
+The command below will mount FloydHub's public
+[Udacity GAN](https://www.floydhub.com/floydhub/datasets/udacity-gan)
+ dataset at `/my_data`
+```
+floyd run --data floydhub/datasets/udacity-gan:my_data "python my_script.py"
```

Comment:
One needs to specify a particular version of the dataset, e.g. floydhub/datasets/udacity-gan/1. 

Lets discuss separately whether defaulting to the latest version of a dataset is OK.